src/pybind/mgr/ssh/module.py

```diff
@@ -500,6 +500,45 @@ def describe_service(self, service_type=None, service_id=None,
                                     node_name=node_name)
         return orchestrator.TrivialReadCompletion(result)
 
+    def service_action(self, action, service_type,
+                       service_name=None,
+                       service_id=None):
+        self.log.debug('service_action action %s type %s name %s id %s' % (
+            action, service_type, service_name, service_id))
+        if action == 'reload':
+            return orchestrator.TrivialReadCompletion(
+                ["Reload is a no-op"])
+        daemons = self._get_services(
+            service_type,
+            service_name=service_name,
+            service_id=service_id)
+        results = []
+        for d in daemons:
+            results.append(self._worker_pool.apply_async(
+                self._service_action, (d.service_type, d.service_instance,
+                                       d.nodename, action)))
+        if not results:
+            n = service_name
+            if n:
+                n += '-*'
+            raise RuntimeError('Unable to find %s.%s%s daemon(s)' % (
```

Comment:
`s/RuntimeError/OrchestratorError/`. OrchestratorError exceptions are errors caused by the user. Everything else is considered an internal orchestrator bug.